### PR TITLE
Add customProperties field to material file

### PR DESCRIPTION
**Pull request by Norihito-Hiruma**
modified:   schemas/material_schema.json

## Describe your changes

In response to the result of the discussion in issue #182, customProperties have been added to the material file schema.

**Some questions to ask**:

I have added an item with reference to glTF's extensions, so please review it.

## Issue ticket number and link

Related issue is #182.

### CODE REVIEW

File: schemas/asset_schema.json [hunk 1]
```diff
@@ -181,6 +181,15 @@
...
       }
+    },
+    "customProperties" : {
+      "type" : "array",
+      "description" : "Names of custom properties used somewhere in this asset.",
+      "items" : {
+        "type" : "string",
+      },
+      "uniqueItems" : true,
+      "default" : []
```
Comment (ClemensLinnhoff) [inline schemas/asset_schema.json:192]:
```suggestion
    "customProperties": {
      "type": "object",
      "description" : "Non-standardized asset properties for custom tools or tool chains.",
      "propertyValues": {
        "type": "string"
      }
    }
```

I would propose to do it like this, as this allows adding any number of key-value pairs as properties, like in this example:

```JSON
"customProperties": {
  "test": "my first custom property",
  "test2": "my second custom property"
}
````

File: schemas/material_schema.json [hunk 2]
```diff
@@ -185,6 +185,18 @@
...
+          "description" : "Custom properties in this material.",
+          "items" : {
+            "type" : "object",
+            "properties" : {
+            },
+            "additionalProperties" : {
+              "type" : "object"
+            }
+          }
         }
```
Comment (ClemensLinnhoff) [inline schemas/material_schema.json:200]:
```suggestion
    "customProperties": {
      "type": "object",
      "description" : "Non-standardized material properties for custom tools or tool chains.",
      "propertyValues": {
        "type": "string"
      }
    }
```

Slight adjustment to be harmonized with the definition in the asset file.
```

**Review by ClemensLinnhoff — CHANGES_REQUESTED** [review]
I propose slight adjustments to the schema definitions.
Would this be okay for you @Norihito-Hiruma?

File: schemas/material_schema.json (hunk 2 above)
Comment (Norihito-Hiruma) [reply schemas/material_schema.json:200]: Thank you, Clemens-san, for checking the content and suggesting corrections.

Excuse me, but what is the definition of json schema propertyValues?
No propertyValues were found in the json schema specification.
[JSON Schema Specification](https://json-schema.org/specification)

Anyway,
- In the asset file, declare a descriptive extension name.
- In the material file, you can set multiple detailed parameters and their values included in the declared extension.
This parameter value can be set to either a string or a number.

Is it correct to understand that?
If my understanding is not wrong, I agree with your suggestion.

Comment (ClemensLinnhoff) [reply schemas/material_schema.json:200]: "propertyValues" is basically the same as the "properties" field in your proposal. It is just a name. We can also change it back to "properties".

> In the asset file, declare a descriptive extension name

In the meeting I proposed that we make it more generic. So you can use the custom properties to describe what kind of extension you use in the materials, but you can also use it for other custom properties directly in the asset file.

Comment (Norihito-Hiruma) [reply schemas/material_schema.json:200]: Clemens-san, Thank you for the explanation.
It seems that I missed what you explained during the meeting. I'm sorry.
I had understand your opinion.
I agree with your amendment suggestion.

**Review by ClemensLinnhoff — APPROVED** [review]

**Pull request merged**